### PR TITLE
feat: add flag to enable price-feeder start as a separate process

### PR DESCRIPTION
**Pull request by leonz789**
update go-ethereum version manually, update price-feeer version

integrate price-feeder into exocored cmd as separate process

## Description

This PR add a new flag '--oracle' to support start price-feeder tool on node start.
Require: 
- download price-feeder binary and move to `homedir/` of exocored
- put price-feeder related config files under `homedir/config/`
  - oracle_feeder.yaml
  - oracle_env_chainlink.yaml

----

Closes #XXX

## Summary by CodeRabbit

- **New Features**
	- Introduced new command-line flags for enhanced configuration of the `start` command, including options for enabling the oracle feeder and setting the validator's mnemonic.
	- Added a pre-run functionality to execute a price feeder command before starting the main application.

- **Dependency Updates**
	- Integrated a new price feeder library and updated several existing dependencies to improve functionality, performance, and security.

### CODE REVIEW

**Comment by MaxMustermann2** [comment]
Please fix the build since a misconfigured path can be used to run a malicious process with potentially escalated privileges.

**Review by bwhour — APPROVED** [review]
add some comment, the overall looks good.

File: cmd/exocored/root.go [hunk 1]
```diff
@@ -132,7 +143,24 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
...
 	)
-
+	startCmd, _, _ := rootCmd.Find([]string{cmdStartName})
+	startCmd.Flags().Bool(flagOracle, false, "enable oracle feeder")
+	startCmd.Flags().String(flagMnemonic, "", "set validator consensus key's mnemonic")
+	preRunE := startCmd.PreRunE
+	// add preRun to run price-feeder first before starting the node
+	startCmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if enableOracle, _ := cmd.Flags().GetBool(flagOracle); enableOracle {
+			mnemonic, _ := cmd.Flags().GetString(flagMnemonic)
```
Comment (bwhour) [inline cmd/exocored/root.go:153]: Check if the **mnemonic** is empty before starting the price-feeder ?

File: cmd/exocored/root.go [hunk 2]
```diff
@@ -132,7 +143,24 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
...
+			clientCtx := cmd.Context().Value(client.ClientContextKey).(*client.Context)
+			cmdFeeder := exec.Command(path.Join(clientCtx.HomeDir, feederBianry), flagFeederConfig, path.Join(clientCtx.HomeDir, confPath, confOracle), flagFeederSource, path.Join(clientCtx.HomeDir, confPath), flagFeederMnemonic, mnemonic, cmdStartName)
+			cmdFeeder.Stdout = os.Stdout
+			cmdFeeder.Stderr = os.Stderr
+			if err := cmdFeeder.Start(); err != nil {
```
Comment (bwhour) [inline cmd/exocored/root.go:160]: Concurrency scenario: 
price-feeder is started asynchronously, the code does not wait for it to initialize before continuing execution. 
Will this cause the main program to start working before price-feeder is ready ?

File: cmd/exocored/root.go (hunk 1 above)
Comment (leonz789) [reply cmd/exocored/root.go:153]: the priority is 
 1. cli: --mnemonic
 2. field mnemonic in config file
 3. privatekey from priv_validator_key.json
So we allow this to be empty

File: cmd/exocored/root.go (hunk 2 above)
Comment (leonz789) [reply cmd/exocored/root.go:160]: Yes it would be possible, but it's fine for price-feeder be ready later than exocored

**Review by TimmyExogenous — APPROVED** [review]

**Review by MaxMustermann2 — APPROVED** [review]

**Review by mikebraver — APPROVED** [review]

**Pull request merged**